### PR TITLE
Feat/stripe

### PR DESCRIPTION
**Pull request by TeddyRoncin**
# Changed payment method from Etupay to Stripe

## Changes

* Route `POST /users/current/cart` now returns the token to pay with Stripe.
* Removed old callback routes for Etupay (`GET/POST /etupay`), and added `POST /stripe/accepted` and `POST /stripe/expired` routes to tell when a cart has been payed or expired.
* Added route `PATCH /admin/items/stripe-sync` to synchronize items in Stripe.

## Breaking changes

Everything related to payments, and cart status.

## What's left to do

- [x] Verify build
- [x] Lint
- [x] Remove console.log / commented code
- [x] Test the expired route
- [x] Instantiate `Stripe` better

### CODE REVIEW

File: src/controllers/users/createCart.ts [hunk 1]
```diff
@@ -180,24 +180,23 @@ export default [
...
       }
 
+      if (
+        cartItems.reduce(
+          (previous, current) => previous + (current.reducedPrice ?? current.price) * current.quantity,
+          0,
+        ) < 0
+      ) {
+        return forbidden(response, ResponseError.BasketCannotBeNegative);
+      }
```
Comment (Remi-pcn) [inline src/controllers/users/createCart.ts:190]: Can you add explanation and why is there a "," at line 186 ?

File: tests/stripe/paymentAcceptedWebhook.test.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+  after(async () => {
+    await database.cart.deleteMany();
+    await database.user.deleteMany();
+  });
+
+  const generateCart = (transactionId: string) => ({
+    object: 'event',
+    type: 'checkout.session.completed',
+    another_field_that_should_be_accepted: 'hello there :)',
```
Comment (Remi-pcn) [inline tests/stripe/paymentAcceptedWebhook.test.ts:29]: ?

File: tests/stripe/paymentAcceptedWebhook.test.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    data: {
+      object: {
+        object: 'checkout.session',
+        id: transactionId,
+        and_put_one_more_here: 3,
```
Comment (Remi-pcn) [inline tests/stripe/paymentAcceptedWebhook.test.ts:34]: ?

File: tests/utils/pdf.test.ts [hunk 4]
```diff
@@ -61,7 +61,7 @@ describe('Tests the PDF utils', () => {
       },
     ]);
 
-    const detailedCart = await updateCart(createdCart.id, 123, TransactionState.paid);
+    const detailedCart = await updateCart(createdCart.id, '124', TransactionState.paid);
```
Comment (Remi-pcn) [inline tests/utils/pdf.test.ts:64]: 123 turned into a 124 (normal?)

**Review by Remi-pcn — COMMENTED** [review]
Tout semble bon apart les quelques trucs qu'on a comment. @DevNono si jamais tu peux jeter un coup d'oeil aussi

File: src/controllers/users/createCart.ts (hunk 1 above)
Comment (TeddyRoncin) [reply src/controllers/users/createCart.ts:190]: eslint ^^

File: tests/utils/pdf.test.ts (hunk 4 above)
Comment (TeddyRoncin) [reply tests/utils/pdf.test.ts:64]: yes

**Pull request closed without merging**